app/src/main/java/org/astraea/producer/Builder.java

```diff
@@ -43,8 +47,24 @@
     return this;
   }
 
+  public Builder<Key, Value> transactionalId(String transactionalId) {
```

Comment:
Is this still required if we always generate random id?

Reply:
No, it is not required now. We don't need to manually set the id.
I will delete it.